app/test/ui/widgets/custom_text_field_test.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:medi_support/ui/widgets/custom_text_field.dart';
+
+void main() {
+  final Completer<void> completerSubmit = Completer<void>();
+  final Completer<void> completerOnTabOutside = Completer<void>();
+  CustomTextField customTextField = CustomTextField(
+    onSubmitted: completerSubmit.complete,
+    onTapOutside: completerOnTabOutside.complete,
+  );
+  testWidgets('CustomTextField renders correctly', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: customTextField,
+        ),
+      ),
+    );
+
+    expect(find.byType(CustomTextField), findsOneWidget);
+  });
+
+  testWidgets('CustomTextField renders correctly whit onSubmitted working',
```

Comment:
**suggestion:** 
```suggestion
  testWidgets('CustomTextField renders correctly with onSubmitted working',
```

and extract into separate test. Call backvalidation should be tested separate